Improve gitignore to exclude more file types

michel-slm opened the pull request:
- Ignore temporary files created by Vim and Emacs
- Ignore HTML files generated from Markdown documentation

michel-slm committed:
Improve gitignore to exclude more file types

- Ignore temporary files created by Vim and Emacs
- Ignore HTML files generated from Markdown documentation

relrod commented on .gitignore:
```diff
@@ -1,2 +1,5 @@
-asknot.html
+*~
```
-1, you should put these in your own global gitignore file (see git's `core.excludesfile` option).

relrod replied:
(The `.html` one is fine here though.)

michel-slm committed:
Incorporate pull request feedback

Keep the broadened wildcard for /*.html but drop the exclusion of
editor temporary files -- those should be per-developer global options.

ralphbean commented:
Excellent.  Thanks!  :)
